From ca09969e7a0afc22ad7373436bb891c654a43edf Mon Sep 17 00:00:00 2001
From: Deepa <g.k.deepa@intel.com>
Date: Mon, 5 Oct 2015 15:50:33 +0530
Subject: [PATCH] SQUASHME! [VPG]: drm/i915: Fix NULL pointer crash due to
 intel_crtc usage

The 'intel_crtc' should always point to a valid 'drm_crtc' as long
as the 'drm_crtc' is not NULL. This avoids unnecessary looping to
find a valid crtc during upfront_link_training and thus avoids
a crash due to NULL pointer.

SQUASHME! - This patch should be squashed into the following existing patch:
Author:    Durgadoss R <durgadoss.r@intel.com>
Date:      Fri, 22 May 2015 14:30:07 +0530
Change-Id: I2ae866b32a0998abcf32826efa2504e203dc8ffa
Subject:   FOR_UPSTREAM [VPG]: drm/i915: Enable Upfront link training
           for type-C DP support

Change-Id: Ifae8bb4864c8edd652b226c68fbe14f81275e7da
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-4819
Signed-off-by: Deepa <g.k.deepa@intel.com>
---
 drivers/gpu/drm/i915/intel_dp.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/i915/intel_dp.c b/drivers/gpu/drm/i915/intel_dp.c
index 5abcb7f..a3e5aa3 100644
--- a/drivers/gpu/drm/i915/intel_dp.c
+++ b/drivers/gpu/drm/i915/intel_dp.c
@@ -4755,7 +4755,7 @@ intel_dp_detect(struct drm_connector *connector, bool force)
 	u8 sink_irq_vector;
 	enum intel_display_power_domain power_domain;
 	struct edid *edid = NULL;
-	struct intel_crtc *intel_crtc = NULL;
+	struct intel_crtc *intel_crtc = crtc ? to_intel_crtc(crtc) : NULL;
 #ifdef CONFIG_EXTCON
 	struct intel_connector *intel_connector = to_intel_connector(connector);
 #endif
@@ -4805,7 +4805,6 @@ intel_dp_detect(struct drm_connector *connector, bool force)
 		 */
 		if (intel_encoder->connectors_active &&
 						crtc && crtc->enabled) {
-			intel_crtc = to_intel_crtc(crtc);
 			DRM_DEBUG_KMS("Disabling crtc %c for upfront link training\n",
 					pipe_name(intel_crtc->pipe));
 			intel_crtc_control(crtc, false);
-- 
1.9.1

